dev/src/write-batch.ts

```diff
@@ -509,6 +543,51 @@ export class WriteBatch {
     return this.commit_();
   }
 
+  /**
+   * Commits all pending operations to the database and verifies all
+   * preconditions. Fails the entire write if any precondition is not met.
+   *
+   * The writes in the batch are not applied atomically and can be applied out
+   * of order.
+   *
+   * @private
+   */
+  async batchCommit(commitOptions?: {
+    requestTag?: string;
+  }): Promise<BatchWriteResult[]> {
+    this._committed = true;
+    const tag = (commitOptions && commitOptions.requestTag) || requestTag();
+    await this._firestore.initializeIfNeeded(tag);
+
+    const database = this._firestore.formattedName;
+    const request: api.IBatchWriteRequest = {database};
+    const writes = this._ops.map(op => op());
+    request.writes = [];
+
+    for (const req of writes) {
```

Comment:
Suggestion:

```
    const request: api.ICommitRequest = {database, writes: []};
    
    const writes = this._ops.map(op => op());
    for( ...
```

You may also fix in the original code.

Reply:
thanks!